gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java

```diff
@@ -516,30 +530,37 @@ public FSDataOutputStream create(
       long blockSize,
       Progressable progress)
       throws IOException {
+    Stopwatch stopwatch = Stopwatch.createStarted();
     incrementStatistic(GhfsStatistic.INVOCATION_CREATE);
-    checkArgument(hadoopPath != null, "hadoopPath must not be null");
-    checkArgument(replication > 0, "replication must be a positive integer: %s", replication);
-    checkArgument(blockSize > 0, "blockSize must be a positive integer: %s", blockSize);
+    try {
```

Comment:
It maybe more readable and maintainable if we extract main logic in the `*Internal` methods and wrap calls to them in try-catch blocks:

```java
@Override
... create(. . .) {
  Stopwatch stopwatch = Stopwatch.createStarted();
  try {
    createInternal(. . .);
  } finally {
     GhfsTimeStatistic.incrementTimeElapsedMicrosec(
          CREATE, stopwatch.elapsed(TimeUnit.MICROSECONDS));
  }
}
```

Not sure that it will be more readable but an option to consider.

Reply:
Since I used an existing lambda, I think it should be more readable than before.